src/chapters/ldap-auth.xml

```diff
@@ -356,6 +356,17 @@ dn: cn={4}guacConfigGroup,cn=schema,cn=config
                             "<systemitem>uid=user,ou=people,dc=example,dc=net</systemitem>".</para>
                     </listitem>
                 </varlistentry>
+                <varlistentry>
+                    <term><property>ldap-user-attributes</property></term>
+                    <listitem>
+                        <para>The attribute or attributes to retrieve from the LDAP directory.
+                            These attributes are stored as tokens with the prefix "GUAC_ATTR_"
+                            and the name of the attribute appended in uppercase letters.
+                            The value of the token is the value of the attribute in the LDAP directory.
+                            If the attribute has no value in the directory then the token is not saved.
+                            Multiple attributes can be specified here, separated by commas.</para>
```

Comment:
Two things:
- Probably should clarify here that the attribute(s) are for the currently logged-in user.
- Also probably want to address how multi-valued attributes behaved.  This occurred to me at one point during the phase of writing the code, but left my mind, but it should at least be documented.  For example, in general LDAP allows the "mail" attribute to contain multiple values, and at one of my previous employers there was a large portion of the company that had multiple e-mail addresses.  What happens in this case?  There are several LDAP attributes that fall into that same category that can have multiple values, and it should at least be documented what value gets passed into this token (first value, last value, random value, all of them together...?).

Reply:
If the LDAP attribute has multiple values then the token would get all of the values together. I would assume someone using this feature wouldn't use a attribute with multiple values? If they did, I'm not sure what we could change to help them pick which value to use. Do you have an idea?

Reply:
I wouldn't assume that someone would not use an attribute with multiple values - they may either not be aware or not have control of that.

> If they did, I'm not sure what we could change to help them pick which value to use.

I don't know that we'd be able to help control what of the multiple values is used, but, in the `getLDAPAttributes()` method, instead of using

    attr.getStringValue()

we could use:

    attr.getStringValueArray()[0]

Probably have to do a couple of other checks to make sure that there's actually a value there (we don't get a null array or empty value, but probably worth changing the code slightly to do that, and then just document that, if the attribute is multi-valued, it'll return the first value.

Reply:
So it turns out getStringValue() actually returns the the first value of the attribute as a String. You can find it here:
https://www.novell.com/documentation/developer/jldap/jldapenu/api/
So it already returns the first value. I also just tested this and it does return just the first value of the attribute.

Reply:
Sounds good.